### DISCUSSION TITLE
Simulation - webots python library mismatch

### DISCUSSION THREAD

**Discussion by iNavin** [bitcraze/discussions#970] [Q&A] [answered]
Hi,

I was trying to use the webots simulation for crazyFlie to test code before using it on the actual hardware, the controller code in the present in crazyflie_controller_py from the - https://github.com/bitcraze/crazyflie-simulation mentioned in the (https://www.bitcraze.io/documentation/tutorials/getting-started-with-simulation/) is very different, it is not using the SyncCrazyflie or the motion commander classes, is there any example simulator controller code which uses wrapper classes used in the Step-by-Step: Motion Commander(https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_motion_commander/ ) ?

cause when i was porting the motion commander code , i got stuck at SyncCrazyflie(URI, cf=Crazyflie(rw_cache='./cache')) as scf: - for creating the crazyFlie instance , as there is no URI for simulation 

please also direct me to some resources for simulation.

**Comment by knmcguire** [comment, marked as answer]
Sorry for the confusion. 
The simulator and the crazyflie python library are not linked at all at this moment so you have to see them as completely separate. So it means that you will need to use the webots specific API: https://cyberbotics.com/doc/guide/

At one point we would like to make it possible to have a direct link with the crazyflie python library with some kind of socket (like this [example](https://github.com/bitcraze/crazyflie-simulation/tree/main/simulator_files/webots/controllers/crazyflie_controller_py_socket)) but we need time to implement that still and the autumn is swamped for us.

**Comment by llanesc** [comment]
Hi @iNavin, take a look at this discussion https://github.com/orgs/bitcraze/discussions/995 that I recently opened. It's still fairly new and not  totally polished, but it worked for my testing and deployment of CFlib scripts to hardware. It saved me the trouble of a few crashes from small bugs in my CFlib scripts. The only difference is that I implemented it with Gazebo and not Webots. I'm intending to push this to the official repo at some point and would like to have testers find bugs or improvements to the documentation. Hope this helps.

**Reply by iNavin** [reply]
thanks, this helps.
